src/Testing/TypeInferenceTestCase.php

```diff
@@ -135,7 +135,7 @@ public function assertFileAsserts(
 			$variableName = $args[2];
 			$this->assertTrue(
 				$expectedCertainty->equals($actualCertainty),
-				sprintf('Expected %s, actual certainty of variable $%s is %s in %s on line %d.', $expectedCertainty->describe(), $variableName, $actualCertainty->describe(), $file, $args[3]),
+				sprintf('Expected %s, actual certainty of variable %s is %s in %s on line %d.', $expectedCertainty->describe(), $variableName, $actualCertainty->describe(), $file, $args[3]),
```

Comment:
```suggestion
				sprintf('Expected %s, actual certainty of %s is %s in %s on line %d.', $expectedCertainty->describe(), $variableDescription, $actualCertainty->describe(), $file, $args[3]),
```

And then in `$variableDescription` either have `variable $%s`, or `offset %s`...

Also would be nice to somehow test the assertion message here.

Reply:
I applied the feedback. I added an assertion on the error in FileAssertRuleTest. I think it matters there. As the other errors are currently not reporting (which is good) and are only when the test fails?

Reply:
But if you have an idea where to add the extra test, I make the change 😊